src/Aspire.Dashboard/ResourceService/DashboardClient.cs

```diff
@@ -158,42 +164,6 @@ GrpcChannel CreateChannel()
                     LoggerFactory = _loggerFactory,
                     ThrowOperationCanceledOnCancellation = true
                 });
-
-            X509CertificateCollection GetFileCertificate()
-            {
-                Debug.Assert(
-                    _dashboardOptions.ResourceServiceClient.ClientCertificate.FilePath != null,
-                    "FilePath is validated as not null when configuration is loaded.");
-
-                var filePath = _dashboardOptions.ResourceServiceClient.ClientCertificate.FilePath;
-                var password = _dashboardOptions.ResourceServiceClient.ClientCertificate.Password;
-
-                return [new X509Certificate2(filePath, password)];
```

Comment:
"99%" of the reason to use the new X509CertificateLoader over the old constructors is to not load a PKCS#12/PFX when you weren't expecting to.  Since you were expecting to load a PFX, and you're loading from a local file, it's not the end of the world if you continue calling the legacy constructor.  (And since you're only loading it once you don't care about "we made parallel-loading the same PFX better" feature, etc.)

So, while it is forward looking to get off of this ctor and on to the new loader model, it's not "important" right now.